core/src/main/java/io/questdb/cairo/wal/seq/TableTransactionLog.java

```diff
@@ -98,43 +107,47 @@ static TableMetadataChangeLog getTableMetadataChangeLog() {
         return instance;
     }
 
-    long addEntry(int walId, int segmentId, long segmentTxn) {
+    long addEntry(int structureVersion, int walId, int segmentId, int segmentTxn, long timestamp) {
+        txnMem.putInt(structureVersion);
```

Comment:
adding structureVersion will increase the size of txn log quite a bit.
most of the time this will be a redundant field.
i liked the previous version more when we used another field to store it.
however, agree that it is cleaner and nicer to have the getStructureVersion() method on the cursor's API.

maybe we could go for the best of both world's and keep the getStructureVersion() method on the cursor but also take advantage of the unused fields when structure change is logged.
i think something like this in the apply job could work:
```
final int walId = transactionLogCursor.getWalId();
if (walId != TableTransactionLog.STRUCTURAL_CHANGE_WAL_ID) {
    final int segmentId = transactionLogCursor.getSegmentId();
    final long segmentTxn = transactionLogCursor.getSegmentTxn();
    final long seqTxn = transactionLogCursor.getTxn();
    // process wal commit
} else {
    // the below reads from the same place as transactionLogCursor.getSegmentId()
    final int newStructureVersion = transactionLogCursor.getStructureVersion();
    // apply structure change
}
```

Reply:
I put structure version to the record to make the transaction atomic in the `TableTransactionLog`. When structure change is written now, it's the writing of the single long value in the `MAX_TXN_OFFSET` at the end makes transaction committed.

This should maximise our chances to withstand power down during structure change commit.